src/api/apiHelpers.ts

```diff
@@ -131,21 +131,50 @@ type Result =
         | typeof UNKNOWN_ERROR_WHILE_REFRESHING_ACCESS_TOKEN
     }
 
+let refreshedAccessToken: Promise<Result> | null = null
+
+export const cleanRefreshedAccessToken = (): void => {
+  refreshedAccessToken = null
+}
+
+export const refreshAccessToken = async (
+  api: DefaultApi,
+  remainingRetries = 1
+): Promise<Result> => {
+  if (!refreshedAccessToken) {
+    refreshedAccessToken = refreshAccessTokenWithRetriesOnError(api, remainingRetries).then(
+      (result) => {
+        if (result.result) {
+          const token = decodeAccessToken(result.result)
+          if (token) {
+            // exp is the token's expiration date and iat is the token's issue date
+            const lifetimeInMs = (token.exp - token.iat) * 1000
```

Comment:
suggestion (non-blocking): je préfère avoir du code qui s'autoexplique plutot qu'un commentaire

```suggestion
            const tokenExpirationInSeconds = token.exp
            const tokenIssueDateInSeconds = token.iat
            const lifetimeInMs = (tokenExpirationInSeconds - tokenIssueDateInSeconds) * 1000
```